minifi/minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/service/MiNiFiConfigurationChangeListener.java

```diff
@@ -0,0 +1,208 @@
+                    if (!swapConfigFile.delete()) {
+                        logger.warn("The swap file failed to delete after a failed handling of a change. It should be cleaned up manually.");
+                    }
+                    throw e;
+                }
+            }
+        } catch (ConfigurationChangeException e){
+            logger.error("Unable to carry out reloading of configuration on receipt of notification event", e);
+            throw e;
+        } catch (IOException ioe) {
+            logger.error("Unable to carry out reloading of configuration on receipt of notification event", ioe);
+            throw new ConfigurationChangeException("Unable to perform reload of received configuration change", ioe);
+        } finally {
+            IOUtils.closeQuietly(configInputStream);
+            handlingLock.unlock();
+        }
+    }
+
+    @Override
+    public String getDescriptor() {
+        return "MiNiFiConfigurationChangeListener";
+    }
+
+    private void restartInstance() throws IOException {
+        try {
+            runner.reload();
+        } catch (IOException e) {
+            throw new IOException("Unable to successfully restart MiNiFi instance after configuration change.", e);
+        }
+    }
+
+    private void persistBackNonFlowSectionsFromOriginalSchema(byte[] newSchema, Properties bootstrapProperties, File configFile) {
+        try {
+            ConvertableSchema<ConfigSchema> schemaNew = ConfigTransformer
+                .throwIfInvalid(SchemaLoader.loadConvertableSchemaFromYaml(new ByteArrayInputStream(newSchema)));
+            ConfigSchema configSchemaNew = ConfigTransformer.throwIfInvalid(schemaNew.convert());
+            ConvertableSchema<ConfigSchema> schemaOld = ConfigTransformer
+                .throwIfInvalid(SchemaLoader.loadConvertableSchemaFromYaml(new ByteBufferInputStream(runner.getConfigFileReference().get().duplicate())));
+            ConfigSchema configSchemaOld = ConfigTransformer.throwIfInvalid(schemaOld.convert());
+
+            configSchemaNew.setNifiPropertiesOverrides(configSchemaOld.getNifiPropertiesOverrides());
+
+            if (!overrideCoreProperties(bootstrapProperties)) {
+                logger.debug("Preserving previous core properties...");
+                configSchemaNew.setCoreProperties(configSchemaOld.getCoreProperties());
+            }
+
+            if (!overrideSecurityProperties(bootstrapProperties)) {
+                logger.debug("Preserving previous security properties...");
+                configSchemaNew.setSecurityProperties(configSchemaOld.getSecurityProperties());
+            }
+
+            logger.debug("Persisting changes to {}", configFile.getAbsolutePath());
+            SchemaLoader.toYaml(configSchemaNew, new FileWriter(configFile));
+        } catch (Exception e) {
+            logger.error("Loading the old and the new schema for merging was not successful");
```

Comment:
The exception should be logged:
```suggestion
            logger.error("Loading the old and the new schema for merging was not successful", e);
```

Reply:
Added